File: readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <math.h>
#include <ctype.h>
void r(int);
float aver(int,int);

int main(void)
{
    string s = get_string("Text: ");
    int lc = 0, wc = 0, sc = 0;
    int len = strlen(s);
    for(int i = 0 ; i < len ; i++)
    {
        if (isalpha(s[i]))
        {
            lc++;
        }
        else if(s[i] == ' ' )
        {
            wc++;
        }

        else if (i == len -1 )
        {
            if(isalpha(s[i-1]))
            {
                wc++;

            }
        }

        if((s[i-1] != '.' && s[i-1]!= '?' && s[i-1] != '"' && s[i-1] != '!' )&&(s[i] == '.' || s[i] == '?' || s[i] == '!' ))
        {

            sc++;
        }




    }

    double L = aver(lc, wc);
    double S = aver(sc,wc);
    double score = 0.0588 * L - 0.296 * S - 15.8;
    //printf("%d letter(s)\n%d word(s)\n%d sentence(s)\n",lc,wc,sc);
    r((int) round(score));

}

float aver(int letters, int words)
{
    float let = letters, w = words;
    return (float) let * (100.0/w);
}

void r(int d)
{
    if(d>=16)
    {
        printf("Grade 16+\n");
    }
    else if(d<1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n",d);
    }
}

File: readability.py
from cs50 import *
from math import *

def averL(s):
    words = len(s.split(" "))
    #print(f" words = {words}")
    letters = sum(c.isalpha() for c in s)
    return letters * (100.0 / words)

def averS(s):
    count = 0

    for i in range(1,len(s)):
        if s[i-1] != '.' and s[i-1] != '?' and s[i-1] != '!' and (s[i] == '.' or s[i] == '?' or s[i] == '!' ):
            count += 1
    words = len(s.split(" "))
    return count * (100.0/words)




txt = get_string("Text: ")

L = averL(txt)
S = averS(txt)

#print(f"L = {L}, S = {S}")

grade = int(round(0.0588 * L - 0.296 * S - 15.8))

if grade < 1:
    print("Before Grade 1")
elif grade >= 16:
    print("Grade 16+")
else:
    print(f"Grade {grade}")

